plugins/com.google.cloud.tools.eclipse.appengine.deploy/src/com/google/cloud/tools/eclipse/appengine/deploy/AppEngineDeployUtil.java

```diff
@@ -0,0 +1,64 @@
+package com.google.cloud.tools.eclipse.appengine.deploy;
+
+import com.google.gson.Gson;
+import com.google.gson.JsonParseException;
+import com.google.gson.reflect.TypeToken;
+import java.lang.reflect.Type;
+import java.util.List;
+
+/**
+ * A utility class for App Engine deploy.
+ */
+// TODO: move into appengine-core-plugin
+public class AppEngineDeployUtil {
+  /**
+   * Parse the raw json output of the deployment.
+   *
+   * @return an object modeling the output of a deploy command
+   * @throws JsonParseException if unable to extract the deploy output information needed
+   */
+  public static DeployOutput parseDeployOutput(String jsonOutput) throws JsonParseException {
+    Type deployOutputType = new TypeToken<DeployOutput>() {}.getType();
+    DeployOutput deployOutput = new Gson().fromJson(jsonOutput, deployOutputType);
+    if (deployOutput == null
+        || deployOutput.versions == null || deployOutput.versions.size() != 1) {
```

Comment:
I believe it just returning the details of the version that was deployed. But it is returns a list.